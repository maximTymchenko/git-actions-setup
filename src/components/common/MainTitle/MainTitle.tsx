import React, { FC } from 'react';
import SectionTitle from '../SectionTitle';

interface IMainTitle {
    className?: string;
    title: string;
    subtitleF: string;
    subtitleS: string;
    subtitleB: string;
    subtypeB: string;
    type: any;
    subtype: any;
}

const MainTitle: FC = (props: IMainTitle) => {
    return (
        <div>
            <div className="title__wrapper">
                <SectionTitle type="small" title={props.subtitleF} />
                <SectionTitle type="bold" title={props.subtitleB} />
                <SectionTitle type="small" title={props.subtitleS} />
            </div>
            <h1 data-testid="main-title" className={`title title--${props.type}`}>
                {/* <div className="title__wrapper">
                <span className={`title title--${props.subtype}`}>{props.subtitleF}</span>
                <span className={`title title--${props.subtypeB}`}>{props.subtitleB}</span>
                <span className={`title title--${props.subtype}`}>{props.subtitleS}</span>
            </div> */}

                {props.title}
            </h1>
        </div>
    );
};

export default MainTitle;
