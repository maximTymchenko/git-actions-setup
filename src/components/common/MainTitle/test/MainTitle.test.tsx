import React from 'react';
import MainTitle from '../MainTitle';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(<MainTitle subtitle="test" title="test"></MainTitle>);
    expect(getByTestId('main-title')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<MainTitle subtitle="test" title="test"></MainTitle>).toJSON();
    expect(tree).toMatchSnapshot();
});
