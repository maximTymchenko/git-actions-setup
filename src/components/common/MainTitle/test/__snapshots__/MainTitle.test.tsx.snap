// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div>
  <div
    className="title__wrapper"
  >
    <h2
      className="title title--small"
      data-testid="section-title"
    />
    <h2
      className="title title--bold"
      data-testid="section-title"
    />
    <h2
      className="title title--small"
      data-testid="section-title"
    />
  </div>
  <h1
    className="title title--undefined"
    data-testid="main-title"
  >
    test
  </h1>
</div>
`;
